tests/test_metadata.py

```diff
@@ -0,0 +1,390 @@
+            Requires-Python: 2.5
+            Requires-Python: >2.1
+            Requires-Python: >=2.3.4
+            Requires-Python: >=2.5,<2.7
+            Requires-External: C
+            Requires-External: libpng (>=1.5)
+            Project-URL: Bug Tracker, https://github.com/pypa/setuptools/issues
+            Project-URL: Documentation, https://setuptools.readthedocs.io/
+            Project-URL: Funding, https://donate.pypi.org
+            Requires-Dist: pywin32 (>1.0); sys.platform == 'win32'
+            Obsoletes-Dist: pywin31; sys.platform == 'win32'
+            Requires-Dist: foo (1,!=1.3); platform.machine == 'i386'
+            Requires-Dist: bar; python_version == '2.4' or python_version == '2.5'
+            Requires-Dist: baz (>=1,!=1.3); platform.machine == 'i386'
+            Requires-External: libxslt; 'linux' in sys.platform
+            Provides-Extra: docs
+            Description-Content-Type: text/x-rst; charset=UTF-8
+        """,  # based on PEP 345 / PEP 566
+        "2022-01-16": """\
+            Metadata-Version: 2.2
+            Name: BeagleVote
+            Version: 1.0a2
+            Platform: ObscureUnix
+            Platform: RareDOS
+            Supported-Platform: RedHat 7.2
+            Supported-Platform: i386-win32-2791
+            Keywords: dog,puppy,voting,election
+            Description-Content-Type: text/markdown; charset=UTF-8; variant=GFM
+            Author-email: cschuoltz@example.com, snoopy@peanuts.com
+            License: GPL version 3, excluding DRM provisions
+            Requires-Dist: pkginfo
+            Requires-Dist: PasteDeploy
+            Requires-Dist: zope.interface (>3.5.0)
+            Requires-Dist: pywin32 >1.0; sys_platform == 'win32'
+            Requires-Python: >2.6,!=3.0.*,!=3.1.*
+            Requires-External: C
+            Requires-External: libpng (>=1.5)
+            Requires-External: make; sys_platform != "win32"
+            Project-URL: Bug Tracker, http://bitbucket.org/tarek/distribute/issues/
+            Provides-Extra: pdf
+            Requires-Dist: reportlab; extra == 'pdf'
+            Provides-Dist: OtherProject
+            Provides-Dist: AnotherProject (3.4)
+            Provides-Dist: virtual_package; python_version >= "3.4"
+            Obsoletes-Dist: Foo; os_name == "posix"
+            Dynamic: Maintainer
+            Dynamic: Maintainer-email
+
+            This project provides powerful math functions
+            For example, you can use `sum()` to sum numbers:
+
+            Example::
+
+                >>> sum(1, 2)
+                3
+
+        """,  # https://packaging.python.org/en/latest/specifications/core-metadata
+    }
+
+    @pytest.mark.parametrize("spec", PER_VERSION_EXAMPLES.keys())
+    def test_parsing(self, spec: str) -> None:
+        text = bytes(dedent(self.PER_VERSION_EXAMPLES[spec]), "UTF-8")
+        pkg_info = CoreMetadata.from_pkg_info(text)
+        if not pkg_info.dynamic:
```

Comment:
Probably should be a flag on the parameterized data to make sure there isn't a mistake in the actual data compared to what is expected to be tested.